blobstore/src/test/java/org/jclouds/blobstore/integration/internal/BaseContainerIntegrationTest.java

```diff
@@ -561,6 +562,30 @@ public void testDelimiterList() throws InterruptedException {
       }
    }
 
+   @DataProvider
+   public Object[][] getBlobsToEscape() {
+      ImmutableSet<String> testNames = ImmutableSet.of("%20", "%20 ", " %20", " ");
+      Object[][] result = new Object[1][1];
+      result[0][0] = testNames;
+      return result;
+   }
+
+   @Test(dataProvider = "getBlobsToEscape", groups = {"integration", "live"})
+   public void testBlobNameEscaping(ImmutableSet<String> blobNames) throws InterruptedException {
```

Comment:
[nit] `Set` instead of `ImmutableSet` here? We can take care of this during the merge, I think - no need to update the PR.